src/Standards/Generic/Sniffs/ControlStructures/ControlStructuresBracketsNewLineSniff.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * Checks if Control Structures Brackets are on a line by their own
+ *
+ * @author    Greg Sherwood <gsherwood@squiz.net>
+ * @copyright 2006-2018 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\ControlStructures;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+class ControlStructuresBracketsNewLineSniff implements Sniff
+{
+
+    /**
+     * The number of spaces code should be indented.
+     *
+     * @var integer
+     */
+    public $indent = 4;
+
+    /**
+     * A list of tokenizers this sniff supports.
+     *
+     * @var array
+     */
+    public $supportedTokenizers = ['PHP'];
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [
+            T_IF,
+            T_ELSEIF,
+            T_ELSE,
+            T_FOREACH,
+            T_FOR,
+            T_SWITCH,
+            T_DO,
+            T_WHILE,
+            T_TRY,
+            T_CATCH,
+            T_FINALLY,
+        ];
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param integer                     $stackPtr  The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens    = $phpcsFile->getTokens();
+        $errorData = [strtolower($tokens[$stackPtr]['content'])];
+
+        if (isset($tokens[$stackPtr]['scope_opener']) === false) {
+            if ($tokens[$stackPtr]['code'] !== T_WHILE) {
+                $error = 'Possible parse error: %s missing opening or closing brace';
+                $phpcsFile->addWarning($error, $stackPtr, 'MissingBrace', $errorData);
```

Comment:
This sniff should not put a warning here. They might just be using inline control structures, which another sniff can test for and ban.

Reply:
Removed